### ISSUE TITLE
Bug: indent expects the wrong indentation for annotated fields without modifier

### ISSUE THREAD

**Issue by sephiroth-j** [eslint/eslint#15794]
### Environment

Node version: v16.14.2
npm version: v8.5.0
Local ESLint version: v8.13.0 (Currently used)
Global ESLint version: Not found
Operating System: win32 10.0.19044

### What parser are you using?

@typescript-eslint/parser

### What did you do?

In a TypeScript class with an implicit public field with an annotation, eslint expects the wrong indentation. Make the field explicitly public and the expected indentation will be fine.

**Example code**
```ts
class Foo {
	// causes error in eslint 8
	@Annotation()
	implicitPublic: string;

	// does not cause error in eslint 8
	@Annotation()
	public explicitPublic: string;
}
```

<details>
<summary>Configuration</summary>

```
			"extends": [
				"eslint:recommended",
				"plugin:@angular-eslint/base"
			],
			"rules": {
				"no-trailing-spaces": "error",
				"semi": "error",
				"curly": "error",
				"no-undef": "off",
				"no-unused-vars": "off",
				"no-var": "error",
				"no-useless-return": "error",
				"no-useless-concat": "error",
				"no-unneeded-ternary": "error",
				"no-mixed-operators": "error",
				"no-eval": "error",
				"radix": "error",
				"prefer-const": "error",
				"prefer-arrow-callback": "error",
				"eol-last": ["error", "always"],
				"quotes": [
					"error",
					"single",
					{
						"avoidEscape": true,
						"allowTemplateLiterals": true
					}
				],
				"indent": [
					"error",
					"tab",
					{
						"SwitchCase": 1
					}
				]
			}
```
</details>

### What did you expect to happen?

no indentation error

### What actually happened?

error  Expected indentation of 2 tabs but found 1  indent

### Participation

- [ ] I am willing to submit a pull request for this issue.

### Additional comments

worked with version 7.32.0

**Comment by aladdin-add** [comment]
eslint does not support ts, you'll have to ask ts-eslint for help.

afaict, it's a known issue in ts-eslint. related: https://github.com/eslint/eslint/issues/15299
